portal/src/hook/misc.ts

```diff
@@ -0,0 +1,34 @@
+import { useState, useCallback } from "react";
+
+export function useSimpleRPC<T>(
+  f: (...args: any[]) => Promise<T>
+): {
+  loading: boolean;
+  error: Error | null;
+  rpc: (...args: unknown[]) => Promise<T | null>;
+} {
```

Comment:
We should have a better way to type this function.

```typescript
declare function useSimpleRPC<Arg extends any[], Ret>(f: (...args: Arg) => Promise<Ret>): {
    loading: boolean;
    error: unknown;
    g: (...args: Arg) => Promise<Ret>
}
```

Note that `error` should be `unknown`.